src/metabase/api/table.clj

```diff
@@ -79,6 +80,14 @@
         (sync-database/sync-table! updated-table))
       updated-table)))
 
+(defn- format-fields-for-response [resp]
+  (update resp :fields
+          (fn [fields]
+            (map (fn [{:keys [values] :as field}]
+                   (if (seq values)
+                     (update field :values fv/field-values->pairs)
+                     field))
+                 fields))))
```

Comment:
I typically use for when I need filtering or some of the other list comprehension features vs. avoiding an anonymous fn, but I don't have strong opinions on it, I don't mind switching it.